app/lib/features/pins/pages/pins_page.dart

```diff
@@ -18,76 +17,86 @@ class PinsPage extends ConsumerWidget {
+
     return Scaffold(
       backgroundColor: Theme.of(context).colorScheme.neutral,
       body: CustomScrollView(
         slivers: <Widget>[
-          PageHeaderWidget(
-            title: 'Pins',
-            sectionDecoration: const BoxDecoration(
-              gradient: primaryGradient,
-            ),
-            actions: [
-              IconButton(
-                icon: Icon(
-                  Atlas.plus_circle_thin,
-                  color: Theme.of(context).colorScheme.neutral5,
-                ),
-                iconSize: 28,
-                color: Theme.of(context).colorScheme.surface,
-                onPressed: () => context.pushNamed(
-                  Routes.actionAddPin.name,
-                ),
-              ),
-            ],
-            expandedContent: Text(
-              'Pinned items from all the Spaces you are part of',
-              softWrap: true,
-              style: Theme.of(context).textTheme.bodySmall,
-            ),
+          _buildPageHeader(context),
+          _buildPinsGrid(context, pins),
+        ],
+      ),
+    );
+  }
+
+  // pins section header
+  Widget _buildPageHeader(BuildContext context) {
+    return PageHeaderWidget(
+      title: 'Pins',
+      sectionDecoration: const BoxDecoration(
+        gradient: primaryGradient,
+      ),
+      actions: [
+        IconButton(
+          icon: Icon(
+            Atlas.plus_circle_thin,
+            color: Theme.of(context).colorScheme.neutral5,
           ),
-          pins.when(
-            data: (pins) {
-              final widthCount =
-                  (MediaQuery.of(context).size.width ~/ 600).toInt();
-              const int minCount = 2;
-              if (pins.isEmpty) {
-                return const SliverToBoxAdapter(
-                  child: Center(
-                    child: Text('there is nothing pinned yet'),
-                  ),
-                );
-              }
-              return SliverGrid.builder(
-                itemCount: pins.length,
-                gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
-                  crossAxisCount: max(1, min(widthCount, minCount)),
-                  mainAxisExtent: 100,
-                ),
-                itemBuilder: (context, index) {
-                  final pin = pins[index];
-                  return PinListItemById(
-                    pinId: pin.eventIdStr(),
-                    showSpace: true,
-                  );
-                },
-              );
-            },
-            error: (error, stack) => SliverToBoxAdapter(
-              child: Center(
-                child: Text('Loading failed: $error'),
-              ),
-            ),
-            loading: () => const SliverToBoxAdapter(
-              child: Center(
-                child: Text('Loading'),
-              ),
+          iconSize: 28,
+          color: Theme.of(context).colorScheme.surface,
+          onPressed: () => context.pushNamed(
+            Routes.actionAddPin.name,
```

Comment:
I know this wasn't technically changed in this PR but we should probably have the same _hasAnySpacesWithPermission_-check rather than always showing the button. If you don't want to do it in this PR, please take this and make an issue out of it so we can track it.

Reply:
added a FIXME and opening up issue!